apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/util/TagUtil.java

```diff
@@ -38,6 +38,6 @@ public static void tagReturnSpanSpan(final AbstractSpan span, final Map<String,
 
     public static Boolean isReturnTag(String expression) {
         String[] es = expression.split("\\.");
-        return es.length == 2 && "returnedObj".equals(es[0]);
+        return "returnedObj".equals(es[0]);
```

Comment:
Are there possibilities that the method returns a `String` and the user wants to use that as a tag? In such case, `es.length == 1`

Reply:
> Are there possibilities that the method returns a `String` and the user wants to use that as a tag? In such case, `es.length == 1`

Uh, there is a problem here. When returning objects such as numbers, strings, Boolean values, characters, etc., they need to be processed, otherwise the tag value will be null. Whether to handle only common types such as numbers, strings, Boolean values, and characters.

Reply:
> > Are there possibilities that the method returns a `String` and the user wants to use that as a tag? In such case, `es.length == 1`
> 
> Uh, there is a problem here. When returning objects such as numbers, strings, Boolean values, characters, etc., they need to be processed, otherwise the tag value will be null. Whether to handle only common types such as numbers, strings, Boolean values, and characters.

What about simply call `toString` on the `returnedObj`

Reply:
> > > Are there possibilities that the method returns a `String` and the user wants to use that as a tag? In such case, `es.length == 1`
> > 
> > 
> > Uh, there is a problem here. When returning objects such as numbers, strings, Boolean values, characters, etc., they need to be processed, otherwise the tag value will be null. Whether to handle only common types such as numbers, strings, Boolean values, and characters.
> 
> What about simply call `toString` on the `returnedObj`

That's a good idea.